src/solver/simulation/economy.cpp

```diff
@@ -82,14 +82,15 @@ bool Economy::simulationBegin()
                                             numSpace);
 
             auto options = createOptimizationOptions(study);
-            weeklyOptProblems_[numSpace] = Antares::Solver::Optimization::WeeklyOptimization::
-              create(study,
-                     options,
-                     study.parameters.adqPatchParams,
-                     &pProblemesHebdo[numSpace],
-                     numSpace,
-                     resultWriter,
-                     simulationObserver_.get());
+            Optimization::WeeklyOptimization weekOpt(options,
+                                                     &pProblemesHebdo[numSpace],
+                                                     study.parameters.adqPatchParams,
+                                                     numSpace,
+                                                     resultWriter,
+                                                     simulationObserver_.get());
+
+            weeklyOptProblems_.push_back(weekOpt);
```

Comment:
```suggestion
            weeklyOptProblems_.emplace_back(options, /* other args */);
```
See https://en.cppreference.com/w/cpp/container/vector/emplace_back